### PR TITLE
Add `GDSdataset`

### PR DESCRIPTION
**Pull request by KumoLiu**
### Description
part of https://github.com/Project-MONAI/MONAI/issues/6231
Add `GDSDataset`.
After loading with GDS, data lost shape and dtype, so in this PR, add this information manually.

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/data/dataset.py [hunk 1]
```diff
@@ -1510,3 +1512,177 @@ def __init__(
             dfs=dfs, row_indices=row_indices, col_names=col_names, col_types=col_types, col_groups=col_groups, **kwargs
         )
         super().__init__(data=data, transform=transform)
+
+
+class GDSDataset(PersistentDataset):
+    """
+    Re-implementation of the PersistentDataset. GDSDataset enables a direct direct memory access(DMA) data path between
```
Comment (wyli) [inline monai/data/dataset.py:1519]:
```suggestion
    An extension of the PersistentDataset using direct memory access(DMA) data path between
```

File: monai/data/dataset.py [hunk 2]
```diff
@@ -1510,3 +1512,177 @@ def __init__(
...
+                of pre-computed transformed data tensors. The cache_dir is computed once, and
+                persists on disk until explicitly removed.  Different runs, programs, experiments
+                may share a common cache dir provided that the transforms pre-processing is consistent.
+                If `cache_dir` doesn't exist, will automatically create it.
+                If `cache_dir` is `None`, there is effectively no caching.
+            device: target device to put the output Tensor data. Note that only int can be used to
+                specify the gpu to be used.
+            hash_func: a callable to compute hash from data items to be cached.
+                defaults to `monai.data.utils.pickle_hashing`.
+            pickle_module: string representing the module used for pickling metadata and objects,
```
Comment (wyli) [inline monai/data/dataset.py:1553]: this option seems to be removed in this dataset

File: monai/data/dataset.py [hunk 3]
```diff
@@ -1510,3 +1512,177 @@ def __init__(
...
+            for i, _item in enumerate(_item_transformed):
+                for k in _item:
+                    data_hashfile = f"{hashfile}-{k}-{i}"
+                    meta_hash_file_name = f"{hashfile.name}-{k}-meta-{i}"
+                    self._create_new_cache(_item, data_hashfile, meta_hash_file_name)
+        open(hashfile, "a").close()  # store cacheid
+        return _item_transformed
+
+    def _create_new_cache(self, data, data_hashfile, meta_hash_file_name):
+        _item_transformed_meta = data.meta if isinstance(data, MetaTensor) else {}
```
Comment (wyli) [inline monai/data/dataset.py:1659]: this meta dictionary doesn't take much space, how about storing it with `self._meta_cache["meta_hash_file_name"] = copy(data.meta)`?

**Review by wyli — COMMENTED** [review]
thanks, I have a few minor comments inline:

File: monai/data/dataset.py (hunk 2 above)
Comment (KumoLiu) [reply monai/data/dataset.py]: Thanks! Have removed.

File: monai/data/dataset.py (hunk 3 above)
Comment (KumoLiu) [reply monai/data/dataset.py]: Have updated in this [commit](https://github.com/Project-MONAI/MONAI/pull/6778/commits/d3c240e33f66b61d1c63f4d79ded43b80df1753e), could you please help take another look at it, thanks!

**Review by wyli — APPROVED** [review]
Thanks, it looks good to me.

**Pull request merged**